src/main/java/hello/exceptions/APIExceptionHandler.java

```diff
@@ -0,0 +1,142 @@
+package hello.exceptions;
+import com.mongodb.MongoException;
+import com.mongodb.MongoSocketOpenException;
+import com.mongodb.MongoSocketReadException;
+import com.mongodb.MongoTimeoutException;
+import hello.controller.CustomerController;
+import org.apache.log4j.Logger;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+
+
+        @ControllerAdvice
+        public class APIExceptionHandler {
+
+            static Logger log = Logger.getLogger(CustomerController.class);
+
+
+            @ExceptionHandler(value = HttpMessageNotReadableException.class)
+            public ResponseEntity<APIException> handleConverterErrors(HttpMessageNotReadableException e) {
+                HttpStatus badRequest = HttpStatus.NOT_ACCEPTABLE;
+                log.error("exception caught: " + e);
+
+                APIException apiException = new APIException( //handles payload information
+                        "Please enter data in correct format :" ,
+                        badRequest,
+                        ZonedDateTime.now(ZoneId.of("Z")));
+
+                return new ResponseEntity(apiException, HttpStatus.NOT_ACCEPTABLE);
+
+            }
+
+            @ExceptionHandler(value = BadRequestException.class)
+            public ResponseEntity<APIException> handleBadRequest(BadRequestException e) {
+                HttpStatus badRequest = HttpStatus.BAD_REQUEST;
+                log.error("exception caught: " + e);
```

Comment:
Should this be considered 'exceptional' behaviour? If a user enters something they shouldn't, the application isn't at fault

Reply:
Just to clarify, what I mean by this is should we log it as an 'error'?